lib/api/dsl/match/index.js

```diff
@@ -1,51 +1,72 @@
+/*
+ * Kuzzle, a backend software, self-hostable and ready to use
+ * to power modern apps
+ *
+ * Copyright 2017 Kuzzle
+ * mailto: support AT kuzzle.io
+ * website: http://kuzzle.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
 'use strict';
 
-var TestTables = require('./testTables');
+const TestTables = require('./testTables');
 
 /**
  * Matches documents or messages against stored subscriptions
  *
+ * @class Matcher
  * @param {object} store - DSL storage
  * @constructor
  */
-function Matcher (store) {
-  this.store = store;
-  this.matchers = {
-    everything: require('./matchEverything'),
-    equals: require('./matchEquals'),
-    notequals: require('./matchNotEquals'),
-    exists: require('./matchExists'),
-    notexists: require('./matchNotExists'),
-    range: require('./matchRange'),
-    notrange: require('./matchNotRange'),
-    regexp: require('./matchRegexp'),
-    notregexp: require('./matchNotRegexp'),
-    geospatial: require('./matchGeospatial'),
-    notgeospatial: require('./matchNotGeospatial')
-  };
+class Matcher {
+  constructor(store) {
+    this.store = store;
+    this.matchers = {
+      everything: require('./matchEverything'),
+      equals: require('./matchEquals'),
+      notequals: require('./matchNotEquals'),
+      exists: require('./matchExists'),
+      notexists: require('./matchNotExists'),
+      range: require('./matchRange'),
+      notrange: require('./matchNotRange'),
+      regexp: require('./matchRegexp'),
+      notregexp: require('./matchNotRegexp'),
+      geospatial: require('./matchGeospatial'),
+      notgeospatial: require('./matchNotGeospatial')
+    };
+  }
 
-  return this;
-}
+  /**
+   * Matches data against stored subscriptions
+   *
+   * @param {string} index
+   * @param {string} collection
+   * @param {object} data
+   * @return {Array}
+   */
+  match(index, collection, data) {
+    const testTables = new TestTables(this.store.testTables, index, collection);
+    let operand = null;
 
-/**
- * Matches data against stored subscriptions
- *
- * @param {string} index
- * @param {string} collection
- * @param {object} data
- * @return {Array}
- */
-Matcher.prototype.match = function match (index, collection, data) {
-  var
-    operand = null,
-    testTables = new TestTables(this.store.testTables, index, collection);
+    while ((operand = pickOperand(this.store.foPairs, index, collection, operand)) !== undefined) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Split this 99 characters long line (which is greater than 80 authorized). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3ALineLength)